# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //foundation/aafwk/standard git repository of OpenHarmony 3.1 codebase.

Fix implicit cast of nullptr to boolean value not supported by Clang 14.

Signed-off-by: Esben Haabendal <esben.haabendal@huawei.com>
Upstream-Status: Inappropriate [configuration/integration]

diff --git a/foundation/aafwk/standard/services/abilitymgr/src/call_container.cpp b/foundation/aafwk/standard/services/abilitymgr/src/call_container.cpp
index 4657a6051f27..91d295f28857 100644
--- a/foundation/aafwk/standard/services/abilitymgr/src/call_container.cpp
+++ b/foundation/aafwk/standard/services/abilitymgr/src/call_container.cpp
@@ -76,8 +76,8 @@ std::shared_ptr<CallRecord> CallContainer::GetCallRecord(const sptr<IAbilityConn
 bool CallContainer::RemoveCallRecord(const sptr<IAbilityConnection> & connect)
 {
     HILOG_DEBUG("call container release call record by callback.");
-    CHECK_POINTER_AND_RETURN(connect, nullptr);
-    CHECK_POINTER_AND_RETURN(connect->AsObject(), nullptr);
+    CHECK_POINTER_AND_RETURN(connect, false);
+    CHECK_POINTER_AND_RETURN(connect->AsObject(), false);
 
     auto iter = callRecordMap_.find(connect->AsObject());
     if (iter != callRecordMap_.end()) {
